src/main/java/sorald/processor/SoraldAbstractProcessor.java

```diff
@@ -49,15 +54,45 @@
 		} catch (Exception e) {
 			e.printStackTrace();
 		}
+		return this;
 	}
 
-	public abstract JavaFileScanner getSonarCheck();
+	public SoraldAbstractProcessor initResource(List<Node> segment) throws Exception{
```

Comment:
This method contains several duplicate parts from `initResource(String originalFilesPath)`, but I will fix it myself in a further PR.

Reply:
Ok, I'll leave that to you then. Thanks.